app/controllers/samples_controller.rb

```diff
@@ -124,40 +124,111 @@ def index
   end
 
   def index_v2
-    only_library = ActiveModel::Type::Boolean.new.cast(params[:onlyLibrary])
-    exclude_library = ActiveModel::Type::Boolean.new.cast(params[:excludeLibrary])
+    domain = params[:domain]
+    include_ids = ActiveModel::Type::Boolean.new.cast(params[:includeIds])
+
     limit = params[:limit] ? params[:limit].to_i : MAX_PAGE_SIZE_V2
     offset = params[:offset].to_i
 
-    @samples = if only_library
-                 current_power.library_samples
-               elsif exclude_library
-                 # TODO: (gdingle): does this actually exclude the user's library???
-                 Sample.public_samples
-               else
-                 current_power.samples
-               end
+    samples = samples_by_domain(domain)
+    samples = filter_samples(samples, params)
+
+    limited_samples = samples.offset(offset).limit(limit)
+
+    limited_samples_json = limited_samples.as_json(
+      only: [:id, :name, :sample_tissue, :host_genome_id, :project_id, :created_at, :public],
+      methods: []
+    )
+    samples_visibility = visibility(limited_samples)
 
+    details_json = format_samples(limited_samples).as_json()
+    limited_samples_json.zip(details_json, samples_visibility).map do |sample, details, visibility|
+      sample[:public] = visibility
+      sample[:details] = details
+    end
+
+    results = { samples: limited_samples_json }
+    results[:all_samples_ids] = samples.pluck(:id) if include_ids
```

Comment:
so I guess this will cause two queries to be run
* all samples
* limited samples

and I guess you found that running two queries and returning less data to the client is more efficient than running one query and limiting in the client

Reply:
By the way, I find `include_ids` misleading because it implies that when it is false there are no ids. why not `include_all_samples_ids` as a flag?

Reply:
1st comment: This is done for paginated loaders and as you see is fully configurable (I might constrain the limit value in the future). Client can load all ids or not, or ask for all data.

Sending all ids is actually a legacy behavior, which we might change in the future. It is used for the select all logic for infinite tables

2nd comment: Agree it is misleading. Will rename.